### PR TITLE
[devdoc] Add section “Update the version number of a dependency”

### PR DESCRIPTION
**Pull request by inkydragon**
- [x] add a checklist
- [ ] steps to update jll

### CODE REVIEW

**Comment by ViralBShah** [comment]
Merging this since it is already good. Second piece can come in separate PR for JLLs.

**Review by ViralBShah — APPROVED** [review]

File: doc/src/devdocs/build/build.md [hunk 1]
```diff
@@ -303,3 +303,49 @@ From this point, you should
...
+
+### Example: `OpenLibm`
+
+1. Update Version numbers in `deps/openlibm.version`
+    - `OPENLIBM_VER := 0.X.Y`
+    - `OPENLIBM_BRANCH = v0.X.Y`
+    - `OPENLIBM_SHA1 = new-sha1-hash`
+2. Update Version number in `stdlib/OpenLibm_jll/Project.toml`
+    - `version = "0.X.Y+0"`
+3. Update checksum in `deps/checksums/openlibm`
```
Comment (giordano) [inline doc/src/devdocs/build/build.md:347]: Shouldn't this explain _how_ to update them (`make -f contrib/refresh_checksums.mk openlibm`)?

**Comment by giordano** [comment]
This pull request was merged with a failing "Check" job

**Comment by ViralBShah** [comment]
Submitting a PR to fix the whitespace.

**Pull request merged**